mm/2s2h/Enhancements/Songs/SkipScarecrowSong.cpp

```diff
@@ -21,7 +22,16 @@ void RegisterSkipScarecrowSong() {
          */
         if ((enKakasi->picto.actor.xzDistToPlayer < enKakasi->songSummonDist) &&
             ((BREG(1) != 0) || (gPlayState->msgCtx.ocarinaMode == OCARINA_MODE_ACTIVE))) {
-            *should = true;
+
+            // In Rando we may utilize Ocarina Buttons, ensure this is honored.
+            if (IS_RANDO) {
+                if (Rando::Logic::canPlaySong(OCARINA_SONG_SCARECROW_SPAWN)) {
+                    *should = true;
+                }
+            } else {
+                *should = true;
+            }
```

Comment:
This should bail early and *not* set should to true if the player has insufficient notes. As it is now, if the player uses the Ocarina in range of the scarecrow but lacks sufficient notes, they will be soft locked in the Ocarina playing state.
```suggestion
            // In Rando we may utilize Ocarina Buttons, ensure this is honored.
            if (IS_RANDO && Rando::Logic::canPlaySong(OCARINA_SONG_SCARECROW_SPAWN)) {
                    return;
            }

            *should = true;
```